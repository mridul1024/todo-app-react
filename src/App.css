@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@200&display=swap");

.App {
  font-family: "Poppins", sans-serif;
}

.task-list-header {
  margin: 0;
  padding: 0;
  font-size: 36px;
  margin-top: 20px;
  margin-left: 30%;
}

.App ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.App ul .no-task-text {
  margin: 0;
  padding: 0;
  font-size: 18px;
  margin-top: 20px;
  margin-left: 30%;
}
