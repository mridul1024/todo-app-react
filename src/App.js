// import { React, Component } from "react";
import TaskEntry from "./components/TaskEntry";
import TaskList from "./components/TaskList";
import "./App.css";

import { React, useState } from "react";

export default function App() {
  const [task, settask] = useState([]);

  const addTask = () => {
    let newTask = document.getElementById("task-entry").value;
    if (newTask !== "") {
      settask([
        ...task,
        {
          taskTitle: document.getElementById("task-entry").value,
          currentStatus: "0",
        },
      ]);
    }
  };

  return (
    <div className="App">
      <TaskEntry
        InsertTask={() => {
          addTask();
        }}
      ></TaskEntry>
      <p className="task-list-header">
        Tasks <img src="../down_arrow_icon.png" alt="down-arrow" />
      </p>

      <ul data-testid="task-list-ul">
        {task.length === 0 ? (
          <p className="no-task-text">Your tasks will show up here</p>
        ) : (
          task.map((element, index) => (
            <li key={index.toString()} data-testid="task-list-li">
              <TaskList
                task={element.taskTitle}
                taskStatus={element.currentStatus}
                changeStatus={() => {
                  const newTasks = task.slice();
                  newTasks[index].currentStatus === "0"
                    ? (newTasks[index].currentStatus = "1")
                    : (newTasks[index].currentStatus = "0");
                  settask(newTasks);
                }}
              ></TaskList>
            </li>
          ))
        )}
      </ul>
    </div>
  );
}
