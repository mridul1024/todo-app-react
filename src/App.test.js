import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";

// test("renders <App /> component", () => {
//   render(<App />);
//   const headerElement = screen.getByText(/Todo Application/i);
//   expect(headerElement).toBeInTheDocument();
// });

describe("<App /> component tests ->", () => {
  test("should render a <App /> component and match snapshot", () => {
    let appComponent;
    appComponent = render(<App />);
    expect(appComponent.baseElement).toMatchSnapshot("baseElement");
  });

  //! incomplete test - complete this ASAP
  test("should contain a <TaskEntry /> component", () => {
    let { getByPlaceholderText } = render(<App />);
    expect(
      getByPlaceholderText("Write your new task here !")
    ).toBeInTheDocument();
  });

  test("should contain an unordered list with data-testid=task-list-ul", () => {
    let ul;
    render(<App />);
    ul = screen.getByTestId("task-list-ul");
    expect(ul).toBeInTheDocument();
  });

  test("should contain a list item li with data-testid=task-list-li when a task is entered", () => {
    let li;
    let taskEntryInput;
    let taskEntryButton;
    render(<App />);
    taskEntryInput = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryInput, { target: { value: "Random task" } });
    fireEvent.click(taskEntryButton);
    // let li;
    // render(<App />);
    li = screen.getByTestId("task-list-li");
    expect(li).toBeInTheDocument();
  });

  test("should contain a TaskList component when a task is entered", () => {
    let taskEntryInput;
    let taskEntryButton;
    // let { getByText } = Arender(<App />);
    render(<App />);
    taskEntryInput = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryInput, {
      target: { value: "Random task inside a list" },
    });
    fireEvent.click(taskEntryButton);
    let taskListComponent = screen.getByText("Random task inside a list");
    // let { getByText } = render(<App />);
    expect(taskListComponent).not.toBe(null);
  });

  test("should render 2 list elements when 2 tasks are entered", () => {
    //Arrange
    let taskEntryInput;
    let taskEntryButton;
    let taskUl;
    render(<App />);
    taskEntryInput = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    //Act
    fireEvent.change(taskEntryInput, { target: { value: "task one" } });
    fireEvent.click(taskEntryButton);
    fireEvent.change(taskEntryInput, { target: { value: "task two" } });
    fireEvent.click(taskEntryButton);
    taskUl = screen.getByTestId("task-list-ul");
    //Assert
    expect(taskUl.children.length).toBe(2);
  });

  test("should not display a new task when the input field is empty and add task button is clicked", () => {
    let ul;
    let taskEntryInput;
    let taskEntryButton;
    render(<App />);
    taskEntryInput = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryInput, { target: { value: "" } });
    fireEvent.click(taskEntryButton);
    ul = screen.getByText("Your tasks will show up here");
    expect(ul).not.toBe(undefined);
  });

  test('should change task-list-checkbox src to "../task_complete_checkbox.png" after clicking on task', () => {
    //Arrange
    let taskEntryField, taskEntryButton, taskListItem, taskListCheckBox;
    render(<App />);
    //Act
    taskEntryField = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryField, { target: { value: "new task" } });
    fireEvent.click(taskEntryButton);
    taskListItem = screen.getByTestId("task-list-checkbox");
    fireEvent.click(taskListItem);
    taskListCheckBox = screen.getByTestId("task-list-checkbox");
    //Assert
    expect(taskListCheckBox.getAttribute("src")).toBe(
      "../task_complete_checkbox.png"
    );
  });

  test('should change task-list-placeholder css class to "complete-task" after clicking on task', () => {
    //Arrange
    let taskEntryField, taskEntryButton, taskListItem, taskListText;
    render(<App />);
    //Act
    taskEntryField = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryField, { target: { value: "new task" } });
    fireEvent.click(taskEntryButton);
    taskListItem = screen.getByTestId("task-list-checkbox");
    fireEvent.click(taskListItem);
    taskListText = screen.getByTestId("task-list-placeholder");
    //Assert
    expect(taskListText.getAttribute("class")).toBe("complete-task");
  });

  test('should change task-list-checkbox src to "../task_incomplete_checkbox.png"after clicking on task 2 times', () => {
    //Arrange
    let taskEntryField, taskEntryButton, taskListItem, taskListCheckBox;
    render(<App />);
    //Act
    taskEntryField = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryField, { target: { value: "new task" } });
    fireEvent.click(taskEntryButton);
    taskListItem = screen.getByTestId("task-list-checkbox");
    fireEvent.click(taskListItem);
    fireEvent.click(taskListItem);
    taskListCheckBox = screen.getByTestId("task-list-checkbox");
    //Assert
    expect(taskListCheckBox.getAttribute("src")).toBe(
      "../task_incomplete_checkbox.png"
    );
  });

  test('should change task-list-placeholder css class to "incomplete-task" after clicking task 2 times', () => {
    //Arrange
    let taskEntryField, taskEntryButton, taskListItem, taskListText;
    render(<App />);
    //Act
    taskEntryField = screen.getByTestId("task-entry-test-id");
    taskEntryButton = screen.getByTestId("add-task-button-test-id");
    fireEvent.change(taskEntryField, { target: { value: "new task" } });
    fireEvent.click(taskEntryButton);
    taskListItem = screen.getByTestId("task-list-checkbox");
    fireEvent.click(taskListItem);
    fireEvent.click(taskListItem);
    taskListText = screen.getByTestId("task-list-placeholder");
    //Assert
    expect(taskListText.getAttribute("class")).toBe("incomplete-task");
  });
});
