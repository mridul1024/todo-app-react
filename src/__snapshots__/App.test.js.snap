// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> component tests -> should render a <App /> component and match snapshot: baseElement 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="task-entry-container"
      >
        <input
          data-testid="task-entry-test-id"
          id="task-entry"
          placeholder="Write your new task here !"
          type="text"
        />
        <img
          alt="add_task_button"
          data-testid="add-task-button-test-id"
          src="../add_task_button.png"
        />
      </div>
      <p
        class="task-list-header"
      >
        Tasks 
        <img
          alt="down-arrow"
          src="../down_arrow_icon.png"
        />
      </p>
      <ul
        data-testid="task-list-ul"
      >
        <p
          class="no-task-text"
        >
          Your tasks will show up here
        </p>
      </ul>
    </div>
  </div>
</body>
`;
